vaadin-icons-flow-parent/vaadin-icons-flow/src/test/java/com/vaadin/flow/component/icon/tests/FontIconTest.java

```diff
@@ -0,0 +1,111 @@
+    public void constructor_hasFont() {
+        var icon = new FontIcon("fa-solid", "fa-user");
+        Assert.assertEquals("fa-solid fa-user",
+                String.join(" ", icon.getFont()));
+        Assert.assertEquals("fa-solid fa-user",
+                icon.getElement().getProperty("font"));
+    }
+
+    @Test
+    public void emptyConstructorArgs_hasNoFont() {
+        var icon = new FontIcon();
+        Assert.assertArrayEquals(new String[0], icon.getFont());
+        Assert.assertNull(icon.getElement().getProperty("font"));
+    }
+
+    @Test
+    public void setFont_hasFont() {
+        var icon = new FontIcon();
+        icon.setFont("fa-solid", "fa-user");
+        Assert.assertEquals("fa-solid fa-user",
+                String.join(" ", icon.getFont()));
+        Assert.assertEquals("fa-solid fa-user",
+                icon.getElement().getProperty("font"));
+    }
+
+    @Test
+    public void modifyFont_hasModifiedFont() {
+        var icon = new FontIcon();
+        icon.setFont("fa-solid", "fa-user");
+        icon.setFont("fa-solid");
+        Assert.assertEquals("fa-solid", String.join(" ", icon.getFont()));
+        Assert.assertEquals("fa-solid", icon.getElement().getProperty("font"));
+    }
+
+    @Test
+    public void clearFont_hasNoFont() {
+        var icon = new FontIcon();
+        icon.setFont("fa-solid", "fa-user");
+        icon.setFont();
+        Assert.assertArrayEquals(new String[0], icon.getFont());
+        Assert.assertNull(icon.getElement().getProperty("font"));
+    }
+
+    @Test
+    public void setFontFamily_hasFontFamily() {
+        var icon = new FontIcon();
+        icon.setFontFamily("lumo-icons");
+        Assert.assertEquals("lumo-icons", icon.getFontFamily());
+        Assert.assertEquals("lumo-icons",
+                icon.getElement().getProperty("fontFamily"));
+    }
+
+    @Test
+    public void clearFontFamily_hasNoFontFamily() {
+        var icon = new FontIcon();
+        icon.setFontFamily("lumo-icons");
+        icon.setFontFamily(null);
+        Assert.assertNull(icon.getFontFamily());
+        Assert.assertNull(icon.getElement().getProperty("fontFamily"));
+    }
+
+    @Test
+    public void setCharCode_hasCharCode() {
+        var icon = new FontIcon();
+        icon.setCharCode("\uea0e");
+        Assert.assertEquals("\uea0e", icon.getCharCode());
+        Assert.assertEquals("\uea0e", icon.getElement().getProperty("char"));
+    }
+
+    @Test
+    public void clearCharCode_hasNoCharCode() {
+        var icon = new FontIcon();
+        icon.setCharCode("\uea0e");
+        icon.setCharCode(null);
+        Assert.assertNull(icon.getCharCode());
+        Assert.assertNull(icon.getElement().getProperty("char"));
+    }
+
+    @Test
+    public void setColor_hasColor() {
+        var icon = new FontIcon();
+        icon.setColor("red");
+        Assert.assertEquals("red", icon.getColor());
+        Assert.assertEquals("red", icon.getElement().getStyle().get("color"));
+    }
```

Comment:
Should there also be a test that checks that calling `icon.setColor(null)` removes the color?

Reply:
Added